docs/azure/sdk/unit-testing-mocking.md

```diff
@@ -0,0 +1,348 @@
+        var clientMock = new Mock<SecretClient>();
+        clientMock.Setup(c => c.GetPropertiesOfSecretsAsync(It.IsAny<CancellationToken>()))
+            .Returns(pageable);
+    
+        // Create an instance of a class to test passing in the mock client
+        var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), clientMock.Object);
+    
+        // Act
+        var soonToExpire = await finder.GetAboutToExpireSecrets();
+    
+        // Assert
+        Assert.Empty(soonToExpire);
+    }
+
+    [Fact]
+    public async Task ReturnsSecretsThatExpireSoon()
+    {
+        // Arrange
+    
+        // Create a page of enumeration results
+        DateTimeOffset now = DateTimeOffset.Now;
+        Page<SecretProperties> page = Page<SecretProperties>.FromValues(new[]
+        {
+            new SecretProperties("secret1") { ExpiresOn = now.AddDays(1) },
+            new SecretProperties("secret2") { ExpiresOn = now.AddDays(2) },
+            new SecretProperties("secret3") { ExpiresOn = now.AddDays(3) }
+        }, null, Mock.Of<Response>());
+    
+        // Create a pageable that consists of a single page
+        AsyncPageable<SecretProperties> pageable = AsyncPageable<SecretProperties>.FromPages(new [] { page });
+    
+        // Setup a client mock object to return the pageable when GetPropertiesOfSecretsAsync is called
+        var clientMock = new Mock<SecretClient>();
+        clientMock.Setup(c => c.GetPropertiesOfSecretsAsync(It.IsAny<CancellationToken>()))
+            .Returns(pageable);
+    
+        // Create an instance of a class to test passing in the mock client
+        var finder = new AboutToExpireSecretFinder(TimeSpan.FromDays(2), clientMock.Object);
+    
+        // Act
+        var soonToExpire = await finder.GetAboutToExpireSecrets();
+    
+        // Assert
+        Assert.Equal(new[] {"secret1", "secret2"}, soonToExpire);
+    }
+}
+```
+
+## Refactoring your types for testability
+
+Classes that need to be tested should be designed with the ability to replace their dependency implementations. It was a seamless process to replace the `SecretClient` implementation in the example from the previous section because it was one of the constructor parameters. However, there may be classes in your code that create their own dependencies and are not easily testable, such as the following:
+
+```csharp
+public class AboutToExpireSecretFinder
+{
+    public AboutToExpireSecretFinder(TimeSpan threshold)
+    {
+        _threshold = threshold;
+        _client = new SecretClient(new Uri(Environment.GetEnvironmentVariable("KeyVaultUri")), new DefaultAzureCredential());
+    }
+}
+```
+
+The simplest refactoring you can do to enable testing with dependency injection would be to expose the client as a parameter and run default creation code when no value is provided. This approach allows you to make the class testable while still retaining the flexibility of using the type without much ceremony.
```

Comment:
Might make sense to add a cross reference article link to the DI with Azure content.